### ISSUE TITLE
Desktop: System.IO.Tests.WaitForChangedTests.CreatedDeleted_Success failed with "Xunit.Sdk.EqualException"

### ISSUE THREAD

**Issue by Jiayili1** [dotnet/corefx#17728]
Configuration: outerloop_netfx_windows_nt_debug ([build#1](https://ci.dot.net/job/dotnet_corefx/job/master/job/outerloop_netfx_windows_nt_debug/1/testReport/System.IO.MemoryMappedFiles.Tests/MemoryMappedFileTests_CreateFromFile/FileDoesNotExist_mode__Truncate_/))

Failed test: System.IO.Tests.WaitForChangedTests.CreatedDeleted_Success

Message:
~~~
Assert.Equal() Failure\r\nExpected: Deleted\r\nActual: 0
~~~

Stack Trace:
~~~
at System.IO.Tests.WaitForChangedTests.CreatedDeleted_Success(WatcherChangeTypes changeType) in D:\j\workspace\outerloop_net---903ddde6\src\System.IO.FileSystem.Watcher\tests\FileSystemWatcher.WaitForChanged.cs:line 152
~~~

Detail: https://ci.dot.net/job/dotnet_corefx/job/master/job/outerloop_netfx_windows_nt_debug/1/testReport/System.IO.Tests/WaitForChangedTests/CreatedDeleted_Success_changeType__Deleted_/

**Comment by danmosemsft** [comment]
@ianhays is there a relevant difference in the file watcher implementation since desktop?

**Comment by danmosemsft** [comment]
This error indicates the delete change event was not recieved. I verified that increasing the timeout does not help. Is it ever fired, on desktop? I see `WatcherChangeTypes.Deleted` does exist on desktop.

**Comment by danmosemsft** [comment]
@JeremyKuhne @stephentoub  in Ian's absence.

**Comment by safern** [comment]
This was fixed by @ianhays on PR: #17699 

I don't see it failing anymore.
